src/main/java/appeng/core/sync/packets/PacketPatternSlot.java

```diff
@@ -22,79 +22,118 @@
 import java.io.IOException;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.inventory.IInventory;
 
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
+
 import appeng.api.AEApi;
 import appeng.api.storage.data.IAEItemStack;
 import appeng.container.implementations.ContainerPatternTerm;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.util.item.AEItemStack;
 
 
-public class PacketPatternSlot extends AppEngPacket
+public class PacketPatternSlot implements AppEngPacket, AppEngPacketHandler<PacketPatternSlot, AppEngPacket>
 {
 
-	public final IAEItemStack slotItem;
-
-	public final IAEItemStack[] pattern = new IAEItemStack[9];
-
-	public final boolean shift;
+	private IInventory pat;
+	private IAEItemStack slotItem;
+	private IAEItemStack[] pattern = new IAEItemStack[9];
+	private boolean shift;
 
 	// automatic.
-	public PacketPatternSlot( ByteBuf stream ) throws IOException
+	public PacketPatternSlot()
 	{
-
-		this.shift = stream.readBoolean();
-
-		this.slotItem = this.readItem( stream );
-
-		for( int x = 0; x < 9; x++ )
-		{
-			this.pattern[x] = this.readItem( stream );
-		}
 	}
 
-	public IAEItemStack readItem( ByteBuf stream ) throws IOException
+	public IAEItemStack readItem( ByteBuf stream )
 	{
-		boolean hasItem = stream.readBoolean();
+		final boolean hasItem = stream.readBoolean();
 
 		if( hasItem )
 		{
-			return AEItemStack.loadItemStackFromPacket( stream );
+			try
+			{
+				return AEItemStack.loadItemStackFromPacket( stream );
+			}
+			catch( IOException e )
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)